src/main/java/net/minecraftforge/server/command/CommandTreeBase.java

```diff
@@ -143,19 +213,20 @@ public void execute(MinecraftServer server, ICommandSender sender, String[] args
     {
         if (args.length < 1)
         {
-            String subCommandsString = getAvailableSubCommandsString(server, sender);
-            sender.sendMessage(TextComponentHelper.createComponentTranslation(sender, "commands.tree_base.available_subcommands", subCommandsString));
+            sender.sendMessage(TextComponentHelper.createComponentTranslation(sender, "commands.tree_base.available_subcommands", getAvailableSubCommandsString(server, sender)));
         }
         else
         {
             ICommand cmd = getSubCommand(args[0]);
 
-            if(cmd == null)
+            if (cmd == null)
             {
-                String subCommandsString = getAvailableSubCommandsString(server, sender);
-                throw new CommandException("commands.tree_base.invalid_cmd.list_subcommands", args[0], subCommandsString);
+                ITextComponent textComponent = TextComponentHelper.createComponentTranslation(sender, "commands.tree_base.invalid_cmd", args[0]);
+                textComponent.getStyle().setColor(TextFormatting.RED);
+                sender.sendMessage(textComponent);
```

Comment:
I don't think you should be removing the throw. Its up to calling code for how to display the error.

Reply:
But then it would show broken language key on vanilla clients

Reply:
This is not the right place to fix that. Fix the exception handling if you think its an issue.